baseapp/baseapp.go

```diff
@@ -287,6 +290,28 @@ func (app *BaseApp) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 		msg := "application doesn't support queries"
 		return sdk.ErrUnknownRequest(msg).QueryResult()
 	}
+	// Special prefix backslash for special queries
+	path := req.Path
+	if strings.HasPrefix(path, "\\") {
+		query := path[1:]
+		var result sdk.Result
+		switch query {
+		case "simulate":
+			txBytes := req.Data
+			tx, err := app.txDecoder(txBytes)
+			if err != nil {
+				result = err.Result()
+			}
+			result = app.runTx(true, true, txBytes, tx)
```

Comment:
I think we can do this through Query, but we can't just use `runTx` as is because we want a fresh throw-away state to run the simulation against.

If we do it as is, it will update our checktx state, which sounds appealing, but might not be good because then when we try to add it to the tendermint mempool it will call check tx and possibly fail (since the tx was already executed against the checktx state in the app without tendermint knowing).

So we probably need to refactor runTx to take a ctx so we can pass in the ctx we want and create something fresh every time `simulate` is called that just gets thrown away after.

Of course this also means that we cant run muiltiple dependent `simulate` in a row without waiting for block commits.

An alternative option would be to open another connection between Tendermint and App and expose that over the Tendermitn RPC but that would be much more involved ...

@jaekwon